### PR TITLE
Implement the Git REPL mode

### LINKED ISSUE #21: Git REPL mode
Idea [suggested](https://discourse.julialang.org/t/ann-gitcommand-jl-use-command-line-git-in-your-julia-packages/32330/3) by @oschulz: have a Git REPL mode.  [`ReplMaker.jl`](https://github.com/MasonProtter/ReplMaker.jl) could probably come in handy to do this.

### LINKED ISSUE #27: Git REPL mode: disable automatic loading
By defualt, when you do either `import GitCommand` or `using GitCommand`, it automatically loads in the Git REPL mode. Perhaps some users will not want this? So we should have a way of disabling the automatic loading. Probably with an environment variable

### PR DESCRIPTION
**Pull request by DilumAluthge**
Fixes #21 
Fixes #27

### CODE REVIEW

**Comment by DilumAluthge** [comment]
@giordano Want to give the Git REPL mode a try and see what you think?

**Comment by DilumAluthge** [comment]
bors merge

**Comment by DilumAluthge** [comment]
I'm going to merge it into master, but with the version number at a prerelease `0.3.0-DEV`.

**Comment by DilumAluthge** [comment]
bors merge

**Comment by DilumAluthge** [comment]
bors merge

File: Project.toml [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
 uuid = "49b5b516-ca3f-4003-a081-42bdcf55082d"
 authors = ["Dilum Aluthge", "Brown Center for Biomedical Informatics"]
-version = "0.2.0"
+version = "0.3.0-DEV"
 
 [deps]
 Git_jll = "f8c6e375-362e-5223-8a59-34ff63f689eb"
+ReplMaker = "b873ce64-0db9-51f5-a568-4457d8e49576"
 
 [compat]
```
Comment (giordano) [inline Project.toml]: Need to add a compat entry?

File: src/init.jl [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (giordano) [inline src/init.jl]: Did you mean to put some content in this file?

**Comment by DilumAluthge** [comment]
bors merge-

File: src/init.jl (hunk 2 above)
Comment (DilumAluthge) [reply src/init.jl]: Nope it should be deleted.

File: Project.toml (hunk 1 above)
Comment (DilumAluthge) [reply Project.toml]: Yes.

Comment (DilumAluthge) [reply Project.toml]: Yep. Done.

**Comment by DilumAluthge** [comment]
bors merge

File: src/GitCommand.jl [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+export gitrepl
+
+const GIT_REPL_MODE_NAME = "GitCommand.jl Git REPL mode"
+const GIT_REPL_MODE_PROMPT_TEXT = "git> "
+const GIT_REPL_MODE_START_KEY = ','
 
 include("env_mapping.jl")
 include("git_cmd.jl")
+include("git_repl.jl")
+include("init.jl")
```
Comment (giordano) [inline src/GitCommand.jl]: You want to remove also this line probably 😉

**Comment by DilumAluthge** [comment]
bors merge

**Comment by oschulz** [comment]
I tried "using GitCommand" and then "," with #b5db541, but nothing seems to happen. Do I need to do anything else to try this?

**Comment by DilumAluthge** [comment]
> I tried "using GitCommand" and then "," with [b5db541](https://github.com/bcbi/GitCommand.jl/commit/b5db54179d7f97926ed8252de6c6032698e51c93), but nothing seems to happen. Do I need to do anything else to try this?

1. Step 1: `using GitCommand`
2. Step 2: `gitrepl()`
3. Step 3: Press `,` to enter Git REPL mode

**Comment by DilumAluthge** [comment]
Why is CirrusCI refusing to report the `FreeBSD JULIA_VERSION:1.3` status? The build passed. And they reported the `FreeBSD JULIA_VERSION:nightly` status. But I need both to be reported in order for the PR to pass.

**Comment by DilumAluthge** [comment]
bors merge-

**Comment by DilumAluthge** [comment]
bors merge

**Comment by DilumAluthge** [comment]
bors merge

**Comment by DilumAluthge** [comment]
There's clearly some problem with CirrusCI pushing statuses to GitHub.

**Comment by DilumAluthge** [comment]
bors merge

**Pull request merged**